tests/extras/datasets/pandas/test_sql_dataset.py

```diff
@@ -224,12 +233,71 @@ def test_table_exists_mocked(self, mocker, table_data_set):
         self._assert_sqlalchemy_called_once()
 
 
-class TestSQLQueryDataSet:
-    @staticmethod
-    def _assert_pd_called_once():
-        _callable = pd.read_sql_query
-        _callable.assert_called_once_with(sql=SQL_QUERY, con=CONNECTION)
+class TestSQLTableDataSetSingleConnection:
+    def test_single_connection_performance(self, dummy_dataframe):
+        kwargs = dict(
+            save_args=dict(if_exists="append"),
+            table_name=TABLE_NAME,
+            credentials=dict(con=CONNECTION),
+        )
+        datasets = [SQLTableDataSet(**kwargs) for _ in range(10)]
+
+        for d in datasets:
+            d.save(dummy_dataframe)
+
+        start = time.time()
+        for d in datasets:
+            d.load()
+        end = time.time()
+
+        delta = end - start
+        print(delta)
+        cons = datasets[0].engines
+        for con in cons.values():
+            con.dispose()
+
+    def test_single_connection(self, dummy_dataframe, mocker):
```

Comment:
Maybe worth looking into testing the number of connections for the engine as well: https://stackoverflow.com/questions/34775501/how-could-i-check-the-number-of-active-sqlalchemy-connections-in-a-pool-at-any-g#comment114580726_34781703

Reply:
Hmm I looked into this but we seem to be getting a `NullPoll` in our tests, which doesn't have the nice `status()` return. 🤔